pkg/ipam/subnet_test.go

```diff
@@ -0,0 +1,926 @@
+	require.Equal(t, "2001:db8::1", v6Pool.V6IPs.At(0).start.String())
+	require.Equal(t, "2001:db8::ffff:ffff:ffff:fffe", v6Pool.V6IPs.At(0).end.String())
+
+	// check pool V6Free
+	require.Equal(t, 5, v6Pool.V6Free.Len())
+	require.Equal(t, "2001:db8::1", v6Pool.V6Free.At(0).start.String())
+	require.Equal(t, "2001:db8::1", v6Pool.V6Free.At(0).end.String())
+	require.Equal(t, "2001:db8::3", v6Pool.V6Free.At(1).start.String())
+	require.Equal(t, "2001:db8::3", v6Pool.V6Free.At(1).end.String())
+	require.Equal(t, "2001:db8::5", v6Pool.V6Free.At(2).start.String())
+	require.Equal(t, "2001:db8::ff", v6Pool.V6Free.At(2).end.String())
+	require.Equal(t, "2001:db8::101", v6Pool.V6Free.At(3).start.String())
+	require.Equal(t, "2001:db8::251", v6Pool.V6Free.At(3).end.String())
+	require.Equal(t, "2001:db8::255", v6Pool.V6Free.At(4).start.String())
+	require.Equal(t, "2001:db8::ffff:ffff:ffff:fffe", v6Pool.V6Free.At(4).end.String())
+
+	// check pool V6Available
+	require.Equal(t, 5, v6Pool.V6Available.Len())
+	require.Equal(t, "2001:db8::1", v6Pool.V6Available.At(0).start.String())
+	require.Equal(t, "2001:db8::1", v6Pool.V6Available.At(0).end.String())
+	require.Equal(t, "2001:db8::3", v6Pool.V6Available.At(1).start.String())
+	require.Equal(t, "2001:db8::3", v6Pool.V6Available.At(1).end.String())
+	require.Equal(t, "2001:db8::5", v6Pool.V6Available.At(2).start.String())
+	require.Equal(t, "2001:db8::ff", v6Pool.V6Available.At(2).end.String())
+	require.Equal(t, "2001:db8::101", v6Pool.V6Available.At(3).start.String())
+	require.Equal(t, "2001:db8::251", v6Pool.V6Available.At(3).end.String())
+	require.Equal(t, "2001:db8::255", v6Pool.V6Available.At(4).start.String())
+	require.Equal(t, "2001:db8::ffff:ffff:ffff:fffe", v6Pool.V6Available.At(4).end.String())
+
+	// check pool V6Reserved
+	require.Equal(t, 4, v6Pool.V6Reserved.Len())
+	require.Equal(t, "2001:db8::2", v6Pool.V6Reserved.At(0).start.String())
+	require.Equal(t, "2001:db8::2", v6Pool.V6Reserved.At(0).end.String())
+	require.Equal(t, "2001:db8::4", v6Pool.V6Reserved.At(1).start.String())
+	require.Equal(t, "2001:db8::4", v6Pool.V6Reserved.At(1).end.String())
+	require.Equal(t, "2001:db8::100", v6Pool.V6Reserved.At(2).start.String())
+	require.Equal(t, "2001:db8::100", v6Pool.V6Reserved.At(2).end.String())
+	require.Equal(t, "2001:db8::252", v6Pool.V6Reserved.At(3).start.String())
+	require.Equal(t, "2001:db8::254", v6Pool.V6Reserved.At(3).end.String())
+
+	// check pool V6Released
+	require.Equal(t, 0, v6Pool.V6Released.Len())
+
+	// check pool V6Using
+	require.Equal(t, 0, v6Pool.V6Using.Len())
+}
+
+func TestGetV4StaticAddress(t *testing.T) {
+	excludeIps := []string{
+		"10.0.0.2", "10.0.0.4", "10.0.0.100",
+		"10.0.0.252", "10.0.0.253", "10.0.0.254",
+	}
+	subnet, err := NewSubnet("v4Subnet", "10.0.0.0/24", excludeIps)
+	if err != nil {
+		t.Errorf("failed to create v4 stack subnet: %v", err)
+	}
+	// 1. pod1 has v4 ip but no mac, should get specified ip and mac
+	podName := "pod1"
```

Comment:
```suggestion
	podName := "pod1.default"
```

Append namespace suffix to avoid misleading readers.